core/model/modx/processors/browser/file/rename.class.php

```diff
@@ -40,10 +40,21 @@ public function process() {
         }
 
         $oldFile = $this->getProperty('path');
-        $oldFile = preg_replace('/[\.]{2,}/', '', htmlspecialchars($oldFile));
-        $name = $this->getProperty('name');
-        $name = preg_replace('/[\.]{2,}/', '', htmlspecialchars($name));
-        $success = $this->source->renameObject($oldFile, $name);
+        $directory = preg_replace('/[\.]{2,}/', '', htmlspecialchars(pathinfo($oldFile, PATHINFO_DIRNAME)));
+        if (!empty($directory)) {
+            $directory .= DIRECTORY_SEPARATOR;
+        }
+        $name = htmlspecialchars(end(explode( DIRECTORY_SEPARATOR, $oldFile )));
+        $oldFile = $directory.$name;
+
+        $newFile = $this->getProperty('name');
+        $directory = preg_replace('/[\.]{2,}/', '', htmlspecialchars(pathinfo($newFile, PATHINFO_DIRNAME)));
+        if (!empty($directory)) {
+            $directory .= DIRECTORY_SEPARATOR;
+        }
+        $name = htmlspecialchars(end(explode( DIRECTORY_SEPARATOR, $newFile )));
```

Comment:
Same here.

Reply:
@Jako,

my last fix with  [_setlocale_](https://github.com/modxcms/revolution/pull/14330/commits/aad01d99d61a886e51b4f6e8d1b8ebcd400c50b4) is not right yet?